custom/enikshay/tasks.py

```diff
@@ -0,0 +1,180 @@
+import datetime
+from collections import defaultdict
+import pytz
+from xml.etree import ElementTree
+
+from celery.task import periodic_task
+from celery.schedules import crontab
+from django.conf import settings
+from django.utils.dateparse import parse_datetime
+
+from casexml.apps.case.mock import CaseBlock
+from corehq import toggles
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.apps.fixtures.models import FixtureDataItem
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.util.soft_assert import soft_assert
+from dimagi.utils.decorators.memoized import memoized
+
+from .case_utils import get_adherence_cases_between_dates_from_episode, CASE_TYPE_ADHERENCE, CASE_TYPE_EPISODE
+
+
+DOSE_TAKEN_INDICATORS = [
+    'directly_observed_dose',
+    'unobserved_dose',
+    'self_administered_dose',
+]
+DAILY_SCHEDULE_FIXTURE_NAME = 'adherence_schedules'
+DAILY_SCHEDULE_ID = 'schedule_daily'
+SCHEDULE_ID_FIXTURE = 'id'
+
+
+@periodic_task(
+    run_every=crontab(day_of_week=[1], hour=0, minute=0),  # every Monday
+    queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery')
+)
+def enikshay_adherence_task():
+    domains = toggles.UATBC_ADHERENCE_TASK.get_enabled_domains()
+    for domain in domains:
+        updater = PeriodicCaseUpdater(domain)
+        updater.run()
+
+
+class PeriodicCaseUpdater(object):
+
+    def __init__(self, domain):
+        self.domain = domain
+        self.purge_date = pytz.UTC.localize(datetime.datetime.today() - datetime.timedelta(days=60))
```

Comment:
It's mentioned here
https://docs.google.com/a/dimagi.com/document/d/1FjSdLYOYUCRBuW3aSxvu3Z5kvcN6JKbpDFDToCgead8/edit?pli=1&disco=AAAAAv94gJg

I don't honestly understand the project context of purge date, I have tried to follow the spec and write good test cases to validate